ethereum/core/src/main/java/tech/pegasys/teku/core/ForkChoiceUtil.java

```diff
@@ -84,23 +84,11 @@ public static UnsignedLong compute_slots_since_epoch_start(UnsignedLong slot) {
    */
   private static Optional<Bytes32> get_ancestor(
       ForkChoiceStrategy forkChoiceStrategy, Bytes32 root, UnsignedLong slot) {
-    return forkChoiceStrategy
-        .blockSlot(root)
-        .flatMap(
-            blockSlot -> {
-              if (blockSlot.compareTo(slot) > 0) {
-                return get_ancestor(
-                    forkChoiceStrategy,
-                    forkChoiceStrategy.blockParentRoot(root).orElseThrow(),
-                    slot);
-              } else if (blockSlot.equals(slot)) {
-                return Optional.of(root);
-              } else {
-                // root is older than the queried slot, thus a skip slot. Return earliest root prior
-                // to slot.
-                return Optional.of(root);
-              }
-            });
+    Bytes32 parentRoot = root;
+    while (forkChoiceStrategy.blockSlot(parentRoot).get().compareTo(slot) > 0) {
+      parentRoot = forkChoiceStrategy.blockParentRoot(parentRoot).orElseThrow();
+    }
+    return Optional.of(parentRoot);
```

Comment:
This isn't quite right - the `Optional` returned by `forkChoiceStrategy.blockSlot(parentRoot)` may be empty (which would indicate we're importing a block from an invalid fork).

We'll want:
```
Bytes32 parentRoot = root;
    Optional<UnsignedLong> blockSlot = forkChoiceStrategy.blockSlot(parentRoot);
    while (blockSlot.isPresent() && blockSlot.get().compareTo(slot) > 0) {
      parentRoot = forkChoiceStrategy.blockParentRoot(parentRoot).orElseThrow();
      blockSlot = forkChoiceStrategy.blockSlot(parentRoot);
    }
    return blockSlot.isPresent() ? Optional.of(parentRoot) : Optional.empty();
```

Reply:
Given you're probably asleep and I've already written the code, I've pushed this up.  Will get @cemozerr to review. :)